flang/lib/Semantics/check-omp-structure.cpp

```diff
@@ -4550,6 +4550,95 @@ void OmpStructureChecker::Leave(const parser::DoConstruct &x) {
   Base::Leave(x);
 }
 
+void OmpStructureChecker::Enter(const parser::OpenMPInteropConstruct &x) {
+  bool isTargetSyncOccured{false};
+  bool isDependClauseOccured{false};
+  int targetCount{0}, targetSyncCount{0};
+  const auto &dir{std::get<parser::Verbatim>(x.t)};
+  std::list<std::string> ObjectNameList;
+  PushContextAndClauseSets(dir.source, llvm::omp::Directive::OMPD_interop);
+  const auto &clauseList{std::get<parser::OmpClauseList>(x.t)};
+  for (const auto &clause : clauseList.v) {
+    common::visit(
+      common::visitors{
+        [&](const parser::OmpClause::Init &InitClause) {
+          const auto &InteropTypeList{
+              std::get<parser::OmpInitClause::InteropTypes>(InitClause.v.t)};
+          for (auto &InteropTypeVal : InteropTypeList.v) {
+            if (*(parser::Unwrap<parser::InteropType::Kind>(
+                    InteropTypeVal)) ==
+                parser::InteropType::Kind::TargetSync) {
+              isTargetSyncOccured = true;
+              ++targetSyncCount;
+            } else {
+              ++targetCount;
+            }
+            if (targetCount >= 2 || targetSyncCount >= 2) {
```

Comment:
NIT:
This seems confusing, how about?
```suggestion
            if (targetCount > 1 || targetSyncCount > 1) {
```

Reply:
Addressed the comment on [#1c57d1e](https://github.com/swatheesh-mcw/llvm-project/pull/1/commits/1c57d1ea2b8e7095a834689ad5b832e6a2a84c37)